src/all/comicgrowl/src/eu/kanade/tachiyomi/extension/all/comicgrowl/ComicGrowl.kt

```diff
@@ -1,63 +1,330 @@
-import okhttp3.OkHttpClient
+import okhttp3.Interceptor
+import okhttp3.MediaType.Companion.toMediaType
 import okhttp3.Request
+import okhttp3.Response
+import okhttp3.ResponseBody.Companion.toResponseBody
+import org.jsoup.nodes.Document
 import org.jsoup.nodes.Element
+import uy.kohesive.injekt.injectLazy
+import java.io.ByteArrayOutputStream
+import java.text.SimpleDateFormat
+import java.util.Locale
 
-// TODO: get manga status
-// TODO: filter by status
-// TODO: change cdnUrl as a array(upstream)
-class ComicGrowl : GigaViewer(
-    "コミックグロウル",
-    "https://comic-growl.com",
-    "all",
-    "https://cdn-img.comic-growl.com/public/page",
-) {
+class ComicGrowl(
+    override val lang: String = "all",
+    override val baseUrl: String = "https://comic-growl.com",
+    override val name: String = "コミックグロウル",
+    override val supportsLatest: Boolean = false,
+) : ParsedHttpSource() {
 
-    override val publisher = "BUSHIROAD WORKS"
+    private val json: Json by injectLazy()
 
-    override val chapterListMode = CHAPTER_LIST_LOCKED
+    companion object {
+        private const val PUBLISHER = "BUSHIROAD WORKS"
 
-    override val supportsLatest: Boolean = true
+        private val imageUrlRegex by lazy { Regex("^.*?webp") }
 
-    override val client: OkHttpClient =
-        super.client.newBuilder().addInterceptor(::imageIntercept).build()
+        /**
+         * Get cover image url from [element]
+         */
+        private fun getImageFromElement(element: Element): String {
+            val match = imageUrlRegex.find(element.selectFirst("source")!!.attr("data-srcset"))
+                ?: throw Exception("Cover image url not found")
+            return "https:" + match.value
+        }
+
+        private val DATE_PARSER by lazy { SimpleDateFormat("yyyy-MM-dd", Locale.JAPAN) }
+
+        /**
+         * Parse date from string like "3月31日" to UNIX Epoch time.
+         */
+        private fun String.toDate(): Long {
+            return DATE_PARSER.tryParse(this)
+        }
+    }
+
+    override val client = super.client.newBuilder()
+        .addNetworkInterceptor(::imageDescrambler)
+        .build()
 
-    override fun popularMangaRequest(page: Int): Request = GET(baseUrl, headers)
+    override fun popularMangaRequest(page: Int) = GET("$baseUrl/ranking/manga")
 
-    // Show only ongoing works
-    override fun popularMangaSelector(): String = "ul[class=\"lineup-list ongoing\"] > li > div > a"
+    override fun popularMangaNextPageSelector() = null
 
-    override fun popularMangaFromElement(element: Element) = SManga.create().apply {
-        title = element.select("h5").text()
-        thumbnail_url = element.select("div > img").attr("data-src")
-        setUrlWithoutDomain(element.attr("href"))
+    override fun popularMangaSelector() = ".ranking-item"
+
+    override fun popularMangaFromElement(element: Element): SManga {
+        return SManga.create().apply {
+            url = element.selectFirst("a")!!.attr("href")
```

Comment:
Use:
```suggestion
            setUrlWithoutDomain(element.selectFirst("a")!!.absUrl("href"))
```

With relative URLs you don't have to override `mangaDetailsRequest()`. You would use both `setUrlWithoutDomain()` and `.absUrl()` so Jsoup is responsible of parsing the actual path.

Reply:
Didn't know this before, thank you!